samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -216,15 +218,24 @@ public void run() {
     }
 
     try {
-      //initialize JobCoordinator state
+      // initialize JobCoordinator state
       LOG.info("Starting cluster based job coordinator");
 
-      //create necessary checkpoint and changelog streams, if not created
+      // create necessary checkpoint and changelog streams, if not created
       JobModel jobModel = jobModelManager.jobModel();
       MetadataResourceUtil metadataResourceUtil =
           new MetadataResourceUtil(jobModel, metrics);
       metadataResourceUtil.createResources();
 
+      // fan out the startpoints
```

Comment:
For making this feature fully functional in open-source, there has to be startpoint publisher(tool/API) that writes the startpoints in fanout namespace of meta-store. Are we planning on extending CheckpointTool to support it or writing a new tool for it?

Reply:
Fan out is the responsibility of the job coordinator, which is the purpose of this PR. If there is a special use case where a tool/API needs to do fan out directly, the `fanOut` method will write to the namespace directly and the caller does not need to be aware of that namespace.

Reply:
Discussed offline with @dnishimura . 

Actual question was will we develop a tool similar to `CheckpointTool` to enable open source users to publish startpoints of their samza job. Daniel confirmed that the development of sucj a tool is in progress and will be made available soon.